src/main/java/net/minecraftforge/items/IItemHandlerIterator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.items;
+
+import net.minecraft.item.ItemStack;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+
+public interface IItemHandlerIterator extends Iterator<ItemStack>
```

Comment:
A list iterator can serve this function, I believe.

Reply:
i was planning to allow to skip Empty but i dint get that working 
like
default IItemHandlerIterator itemHandlerIterator(boolean skipEmpty)

and i dont want add/set/remove for that you can use https://github.com/loordgek/MinecraftForge/blob/4d2adefed21c2688a9635919610dbf9189eae410/src/main/java/net/minecraftforge/items/IItemHandler.java#L279